monitor/internal/pod/monitor.go

```diff
@@ -179,24 +178,28 @@ func (m *PodMonitor) Run(ctx context.Context) error {
 		}
 	}()
 
-	select {
-	case err := <-errCh:
-		return fmt.Errorf("pod: %s", err.Error())
-	case <-time.After(5 * time.Second):
-		// we give the controller 5 seconds to report back
-		return errors.New("pod: controller did not start within 5s")
-	case <-controllerStarted:
-		m.kubeClient = mgr.GetClient()
-
-		// call ResyncWithAllPods before we return from here
-		// this will block until every pod at this point in time has been seeing at least one `Reconcile` call
-		// we do this so that we build up our internal PU cache in the policy engine,
-		// so that when we remove stale pods on startup, we don't remove them and create them again
-		if err := ResyncWithAllPods(ctx, m.kubeClient, m.resyncInfo, m.eventsCh, m.localNode); err != nil {
-			zap.L().Warn("Pod resync failed", zap.Error(err))
+waitLoop:
```

Comment:
do we want make this a goroutine, so that we don't block the monitor run?

Reply:
No, the whole point here is to block the monitor run. And yes, it is related to the resync. We must have completed the resync once before we can release the monitor.